opm/simulators/linalg/bda/WellContributionsOCLContainer.cpp

```diff
@@ -127,13 +135,42 @@ namespace bda
                                  cl::Local(lmem1), cl::Local(lmem2), cl::Local(lmem2));
     }
 
+
+    void WellContributionsOCLContainer::applyMSWells(cl::Buffer& x, cl::Buffer& y) {
+        cl::Event event;
+        std::vector<cl::Event> events(2);
+
+        // copy vectors x and y from GPU to CPU
+        queue->enqueueReadBuffer(x, CL_FALSE, 0, sizeof(double) * N, x_msw.data(), nullptr, &events[0]);
+        queue->enqueueReadBuffer(y, CL_FALSE, 0, sizeof(double) * N, y_msw.data(), nullptr, &events[1]);
+        event.waitForEvents(events);
+
+        // actually apply MultisegmentWells
+        for(Opm::MultisegmentWellContribution *well: multisegments){
+            well->setReordering(toOrder.data(), true);
```

Comment:
Why are we always using true? There was one version where this was reorder.

Reply:
The reorder seem to always occur in the OpenCL backend. The ILU allways produce two reorder vectors and one of these vectors (`toOrder`) is always passed to the `WellContributionsOCLContainer` class. That is why I left the reordering operator set to `true`. It seemed useless to set a variable that never changes in this context.